### PR TITLE
API-3338 Add hearing type selection to nod

### PR DESCRIPTION
**Pull request by ElleeB**
## Description of change
A Notice of Disagreement Appeal allows the selection of three different types of decision reviews: direct review, evidence submission, or hearing. If hearing is selected, then a hearing type must be selected: virtual hearing, video conference, or central office

At present, the NOD PDF does not provide an option for selecting a hearing type

The code has been updated to allow for the incoming payload to include a hearing type. The schema has been updated to reflect the condition outlined above: if hearing selected, must provide a hearing type.

## Original issue(s)
https://vajira.max.gov/browse/API-3338

## Testing
notice_of_disagreement model spec modified and additional specs written
corresponding json fixtures modified/updated including 'expected_10182_minimum.pdf'

### CODE REVIEW

File: modules/appeals_api/spec/fixtures/valid_10182.json [hunk 1]
```diff
@@ -45,6 +45,7 @@
         "representativesName": "Harry Styles"
       },
       "boardReviewOption": "hearing",
+      "hearingType": "central_office",
```
Comment (ElleeB) [inline modules/appeals_api/spec/fixtures/valid_10182.json:48]: Modified for testing purposes

File: modules/appeals_api/spec/fixtures/valid_10182_minimum.json [hunk 2]
```diff
@@ -16,7 +16,7 @@
         },
         "emailAddressText": "a@a.a"
       },
-      "boardReviewOption": "hearing",
+      "boardReviewOption": "evidence_submission",
```
Comment (ElleeB) [inline modules/appeals_api/spec/fixtures/valid_10182_minimum.json:19]: Modified for testing purposes

**Comment by ElleeB** [comment]
Addressing failed test

File: modules/appeals_api/app/models/appeals_api/notice_of_disagreement.rb [hunk 3]
```diff
@@ -164,14 +173,41 @@ def add_missing_claimant_info_error(field, attribute_name:)
       errors.add attribute_name, "if any claimant info is present, claimant #{field} must also be present"
     end
 
+    def validate_hearing_type_selection
+      return if board_review_hearing_selected? && includes_hearing_type?
+
+      if hearing_type_missing?
+        errors.add :form_data, "if 'boardReviewOption' 'hearing' is selected, a 'hearingType' must be selected"
```
Comment (cilestin) [inline modules/appeals_api/app/models/appeals_api/notice_of_disagreement.rb:180]: Let's give the full path to the `boardReviewOption` and `hearingType` attributes like we do for `#validate_that_at_least_one_set_of_contact_info_is_present`

File: modules/appeals_api/app/models/appeals_api/notice_of_disagreement.rb [hunk 4]
```diff
@@ -164,14 +173,41 @@ def add_missing_claimant_info_error(field, attribute_name:)
...
+      elsif unexpected_hearing_type_inclusion?
+        errors.add :form_data, "if 'boardReviewOption' 'direct_review' or 'evidence_submission' is selected, " \
```
Comment (cilestin) [inline modules/appeals_api/app/models/appeals_api/notice_of_disagreement.rb:182]: Full paths to `boardReviewOption` and `hearingType` here, too.

**Review by cilestin — CHANGES_REQUESTED** [review]

File: modules/appeals_api/app/models/appeals_api/notice_of_disagreement.rb (hunk 3 above)
Comment (ElleeB) [reply modules/appeals_api/app/models/appeals_api/notice_of_disagreement.rb]: Done! Much better thanks.

File: modules/appeals_api/app/models/appeals_api/notice_of_disagreement.rb (hunk 4 above)
Comment (ElleeB) [reply modules/appeals_api/app/models/appeals_api/notice_of_disagreement.rb:182]: Done! Much better thanks.

**Review by cilestin — APPROVED** [review]
👍

**Pull request merged**